### ISSUE TITLE
Is Visual Studio necesseary? (how to avoid vendor lock-in)

### ISSUE THREAD

**Issue by developer239** [davidortinau/WeatherTwentyOne#46]
I have production experience with React Native. Mobile development is fun but React Native has _some_ issues. .NET MAUI seems like a good alternative although I am a bit confused by the lack of documentation and tutorials.

My main problem right now is that every learning resource that I found forces me to use Visual Studio. I know that for Android / iOS development I usually need to use official IDEs. However, when it comes to cross-platform development I would expect more freedom. For example what if I want to use Rider for the development? What if I want to use VSCode? (I noticed that the author of this repository used VSCode)

**Is there any CLI tool available?** Could you please point me in the right direction?

**Example:**
- With React Native I can run iOS applications on MacOS with the following command: `npx react-native run-ios` [I can also specify the device](https://reactnative.dev/docs/running-on-simulator-ios).
- With React Native I can run Android applications on MacOS with the following command: `npx react-native run-ios`. Honestly, I am not sure if I even need Android Studio. I think I do for the build but when it comes to emulator I can use [Genymotion](https://www.genymotion.com/) instead of Android Studio. How do I run .NET MAUI application with Genymotion?

Thank you. Any additional learning resources would be highly appreciated. Especially anything that isn't locked in the Windows ecosystem. 🙌

**Comment by m1nhtu99-hoan9** [comment]
Based on how you framed your question, I assume that you don't have much experiences with DotNet ecosystem. The counterpart of `npx` or `npm` or `yarn` CLI for project management in CSharp is `dotnet` [(official docs here)](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet). 

And yes, you can use Rider as alternative to Visual Studio. JetBrains' progress on Rider support for DotNet MAUI are updated through [their official blogs](https://blog.jetbrains.com/dotnet/tag/maui/).

**Comment by developer239** [comment]
@m1nhtu99-hoan9 Thank you for taking the time to answer my question. I like how you point me in the right direction by linking official docs and the official blog. However, the problem for me - someone who is new to the .ENT ecosystem - is that there aren't any resources that would show me how to use `dotnet` cli to bootstrap and run the project.

**EXAMPLE**
**Creating first .Net MAUI project**
https://stackoverflow.com/questions/65106633/creating-first-net-maui-project

Accepted answer: ...just follow the official documentation ... install visual studio ... use the GUI to create the project ... (basically what you said)

The second-best answer: ...use this CLI command ... only to realize that it doesn't actually generate the project ...

**Comment by developer239** [comment]
@m1nhtu99-hoan9 Obviously, I am new to .NET ecosystem. Thank you for taking the time to answer. I appreciate the links that you shared but that didn't quite tell me what I wanted to know.

**My question:**
Suppose that I installed all necessary services and dependencies. _How do I run this repository (WeatherTwentyOne) only by using CLI?_

**More Importantly:**
_Where do I find the information myself?_ Every tutorial that showed me how to use Maui used Visual Studio.

**Example:**
1. `npx react-native init AwesomeProject` to generate a new project (is basically git clone WeatherTwentyOne)
2. `npx react-native start` to run webserver (is probably not necessary in this case)
3. **`npx react-native run-ios` to run the application in iOS emulator**
4. **`npx react-native run-android` to run the application in android emulator**

**Comment by hxhieu** [comment]
@developer239 at this stage you need at least VS2022 Preview (yet its not out yet) in order to run a MAUI project. It is the prerequisite stated in the docs https://docs.microsoft.com/en-us/dotnet/maui/get-started/installation

So yes, VS is mandatory
